packages/swapper/src/swappers/RelaySwapper/endpoints.ts

```diff
@@ -0,0 +1,178 @@
+  SwapErrorRight,
+  SwapperApi,
+  SwapperDeps,
+  TradeQuote,
+  TradeRate,
+} from '../../types'
+import { TradeQuoteError } from '../../types'
+import { checkSafeTransactionStatus, isExecutableTradeQuote, makeSwapErrorRight } from '../../utils'
+import { relayChainMap } from './constant'
+import { getTradeQuote } from './getTradeQuote/getTradeQuote'
+import { getTradeRate } from './getTradeRate/getTradeRate'
+import { relayService } from './utils/relayService'
+import type { RelayStatus } from './utils/types'
+
+export const relayApi: SwapperApi = {
+  getTradeQuote: async (
+    input: CommonTradeQuoteInput,
+    deps: SwapperDeps,
+  ): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+    if (input.sellAmountIncludingProtocolFeesCryptoBaseUnit === '0') {
+      return Err(
+        makeSwapErrorRight({
+          message: 'sell amount too low',
+          code: TradeQuoteError.SellAmountBelowMinimum,
+        }),
+      )
+    }
+
+    const tradeQuoteResult = await getTradeQuote(input, deps, relayChainMap)
+
+    return tradeQuoteResult
+  },
+  getTradeRate: async (
+    input: GetTradeRateInput,
+    deps: SwapperDeps,
+  ): Promise<Result<TradeRate[], SwapErrorRight>> => {
+    if (input.sellAmountIncludingProtocolFeesCryptoBaseUnit === '0') {
+      return Err(
+        makeSwapErrorRight({
+          message: 'sell amount too low',
+          code: TradeQuoteError.SellAmountBelowMinimum,
+        }),
+      )
+    }
+
+    const tradeRateResult = await getTradeRate(input, deps, relayChainMap)
+
+    return tradeRateResult
+  },
+  getUnsignedEvmTransaction: async ({
+    chainId,
+    from,
+    stepIndex,
+    tradeQuote,
+    assertGetEvmChainAdapter,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+    const currentStep = tradeQuote.steps[stepIndex]
+    if (!currentStep?.relayTransactionMetadata) throw Error('Invalid step index')
+
+    const { to, value, data, gas, maxFeePerGas, maxPriorityFeePerGas } =
+      currentStep.relayTransactionMetadata
+
+    if (to === undefined || value === undefined || data === undefined || gas === undefined) {
+      const undefinedRequiredValues = [to, value, data, gas].filter(value => value === undefined)
+
+      throw Error('undefined required values in swap step', {
+        cause: {
+          undefinedRequiredValues,
+        },
+      })
+    }
+
+    const feeData = await (async () => {
+      if (bnOrZero(maxFeePerGas).isZero() || bnOrZero(maxPriorityFeePerGas).isZero()) {
+        const { average } = await assertGetEvmChainAdapter(
+          currentStep.sellAsset.chainId,
+        ).getGasFeeData()
+
+        return {
+          maxFeePerGas: undefined,
+          maxPriorityFeePerGas: undefined,
+          gasPrice: average.gasPrice,
+        }
+      }
+
+      return {
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+      }
+    })()
```

Comment:
See https://github.com/shapeshift/web/pull/9188#discussion_r2030278200

Reply:
Done in [29fc44d](https://github.com/shapeshift/web/pull/9188/commits/29fc44d8ce158086f2cbec2c3c7ee0dd44302dac)